src/main/java/io/fabric8/vertx/maven/plugin/mojos/SetupMojo.java

```diff
@@ -246,23 +246,43 @@ private boolean addDependencies(Model model) {
             return false;
         }
 
-        boolean result = false;
+        boolean updated = false;
         List<VertxDependency> deps = VertxDependencies.get();
         for (String dependency : this.dependencies) {
             Optional<VertxDependency> optional = deps.stream()
                 .filter(d -> d.labels().contains(dependency.toLowerCase()))
                 .findAny();
 
             if (optional.isPresent()) {
-                getLog().info("Addind dependency " + optional.get().toCoordinates());
+                getLog().info("Adding dependency " + optional.get().toCoordinates());
                 model.addDependency(optional.get().toDependency());
-                result = true;
+                updated = true;
+            } else if (dependency.contains(":")) {
+                // Add it as a dependency
+                // groupId:artifactId:version:classifier
+                String[] segments = dependency.split(":");
+                if (segments.length >= 2) {
+                    Dependency d = new Dependency();
+                    d.setGroupId(segments[0]);
+                    d.setArtifactId(segments[1]);
+                    if (segments.length >= 3  && ! segments[2].isEmpty()) {
+                        d.setVersion(segments[2]);
+                    }
+                    if (segments.length >= 4) {
+                        d.setClassifier(segments[3]);
+                    }
+                    getLog().info("Adding dependency " + d.getManagementKey());
+                    model.addDependency(d);
+                    updated = true;
+                } else {
+                    getLog().warn("Invalid dependency description '" + dependency + "'");
+                }
```

Comment:
Hard to do for a minimum benefit. I would rather let the developer fix its pom afterward if he override a dependency or add an inherited one.

In fact, if he edits the pom and add the parent afterwards, he has the same issue.